### PR TITLE
Rebrand from lazyledger/lazyledger-core -> celestiaorg/celestia-core

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

This PR rebrands from lazyledger-core celestia-core, while also updating to the latest versions of `nmt` and `rsmt2d`. 

This PR is blocked by
- [ ] renaming the github repo to celestia-core

Closes: #419

### CODE REVIEW

**Comment by renaynay** [comment]
I am personally in favour of keeping the name "lazy ledger" for the core protocol.

File: p2p/ipld/net_test.go [hunk 1]
```diff
@@ -52,7 +52,7 @@ func TestDiscovery(t *testing.T) {
 
 func TestWriteDiscoveryReadData(t *testing.T) {
 	logger := log.TestingLogger()
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*10)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*15)
 	defer cancel()
```
Comment (evan-forbes) [inline p2p/ipld/net_test.go:56]: This test passes locally, but failed twice in the CI, so I increased the timeout 50%

**Comment by evan-forbes** [comment]
interesting, so instead of `celestia-core`, we would use `lazyledger`? I like that too! The people "in the weeds" could still refer to the idea of lazyledger, while celestia/celestia-app/celestiaorg would still being able to be accessible to people not in the weeds. We could also switch lazyledger later when the name celestia is more well known.

I can see this introducing a very small amount of confusion when explaining why we have two repos to someone not familiar with blockchains, but think the pros outweigh that con.

Happy to rename in this PR should we reach social consensus.

**Comment by musalbas** [comment]
I think we just need to bite the bullet and fully rebrand to Celestia. A half-way rebrand is too confusing and non-intuitive. This is the main repo of the Celestia project, therefore it should have Celestia in the name, otherwise it's overcomplicating things for newcomers to the project who don't understand or care about the history or context of lazyledger, vs Celestia. "LazyLedger" being more well-known than "Celestia" is a temporary issue, and will not be helped if we just have a half-complete rebrand.

**Comment by evan-forbes** [comment]
> "LazyLedger" being more well-known than "Celestia" is a temporary issue, and will not be helped if we just have a half-complete rebrand.

that's a great point! We definitely do not have social consensus, therefore it shall stay `celestiaorg/celestia-core`.

File: types/share_merging.go [hunk 2]
```diff
@@ -26,10 +26,12 @@ func DataFromSquare(eds *rsmt2d.ExtendedDataSquare) (Data, error) {
 
 	// iterate over each row index
 	for x := uint(0); x < originalWidth; x++ {
-		// iterate over each col index
-		for y := uint(0); y < originalWidth; y++ {
+		// iterate over each share in the original data square
+		row := eds.Row(x)
+		originalWidth := eds.Width() / 2
+
+		for _, share := range row[:originalWidth] {
```
Comment (evan-forbes) [inline types/share_merging.go:33]: Here's the update to the latest rsmt2d api, which is the most meaningful change. I was careful not to change anything else.

**Review by Wondertan — COMMENTED** [review]
While reviewing I was listing all the places where additional renaming is needed but decided to just submit a commit instead [here](https://github.com/celestiaorg/lazyledger-core/pull/478). 

The last _lazy_ thing left is mentions of the LazyLedger prototype.
@musalbas, should we rename the prototype as well?

**Comment by musalbas** [comment]
I think the ADR abbreviation needs to have Celestia in it, otherwise it will be too plain. e.g. consider EIPs, PEPs, BIPs, all have the name of the project in them. Therefore perhaps we should call them CADR (Celestia Architecture Decision Record), or even CIP (Celestia Improvement Proposal).

**Comment by liamsi** [comment]
Starting to think about CIPs / Celestia Improvement Proposals makes a lot of sense! They should live in a separate repo though. I don't think the ADRs inside of celestia/celestia-core need yet "another celestia" in the name.

**Comment by Wondertan** [comment]
@liamsi, agree with you that inside of `celestia-core` we don't need to call `ADRs` as `Celestia-ADRs`.

@musalbas, the question about the prototype is still left unanswered.

**Comment by musalbas** [comment]
The prototype can continue to be called lazyledger-prototype, because it is the implementation of an academic paper (lazyledger), which Celestia is based on.

File: types/share_merging.go [hunk 3]
```diff
@@ -26,10 +26,12 @@ func DataFromSquare(eds *rsmt2d.ExtendedDataSquare) (Data, error) {
 
 	// iterate over each row index
 	for x := uint(0); x < originalWidth; x++ {
-		// iterate over each col index
-		for y := uint(0); y < originalWidth; y++ {
+		// iterate over each share in the original data square
+		row := eds.Row(x)
+		originalWidth := eds.Width() / 2
```
Comment (liamsi) [inline types/share_merging.go:31]: Why do we need to introduce this variable? In line 17 there is already exactly the same definition.

**Review by liamsi — COMMENTED** [review]
Left one comment. But also:
image

File: types/share_merging.go (hunk 3 above)
Comment (evan-forbes) [reply types/share_merging.go]: whoops, accidentally declared that twice! thanks for catching this!

**Review by liamsi — COMMENTED** [review]
This is almost impossible to review properly. @evan-forbes let's sit down together sometime this week and go through the changes (and how they were made).

But in overall thank you and @Wondertan for this but also
image

**Comment by evan-forbes** [comment]
Yeah, that makes sense. I don't think go modules was made for easy re-brands lol

**Comment by liamsi** [comment]
> or even CIP (Celestia Improvement Proposal).
@musalbas CIPS are already a thing: https://github.com/cosmos/cips

**Review by liamsi — APPROVED** [review]
Let's merge this but let's hold off any further renaming until our devnet!!!!

**Pull request merged**